### PR TITLE
feat: adding SSL Toggle to Create Database Modal

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
This is creating a toggle for forcing SSL certification when creating a Database in Superset. It will be changing the encryption field that is in parameters. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Screen Shot 2021-06-03 at 6 45 49 PM

### TESTING INSTRUCTIONS
Go to databases
+database
choose an engine that allows for expanded params/dynamic form (postgres/mysql/redshift)

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 1]
```diff
@@ -165,6 +173,36 @@ const displayField = ({
...
   />
 );
+const forceSSLField = ({
+  isEditMode,
+  changeMethods,
+  db,
+  sslForced,
+}: FieldPropTypes) => (
+  <div css={(theme: SupersetTheme) => infoTooltip(theme)}>
+    <Tooltip title={t('SSL is required for Preset')}>
```
Comment (betodealmeida) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:183]: Is the plan to replace this before merging the branch into master?

Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:183]: wait, I am confused, replace what portion?

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx [hunk 2]
```diff
@@ -165,6 +173,36 @@ const displayField = ({
...
+              checked: true,
+              value: changed,
+            },
+          });
+        }}
+      />
+      <span css={toggleStyle}>SSL</span>
+    </Tooltip>
+    <InfoTooltip
+      tooltip={t('SSL mode "require" will be used')}
```
Comment (betodealmeida) [inline superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:201]: "require" is psycopg2 specific

```suggestion
      tooltip={t('SSL will be enabled in the database connection')}
```

Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:201]: the figma currently says require. I like this wording better though.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx (hunk 1 above)
Comment (betodealmeida) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:183]: We can't say "SSL is required for Preset" in OSS Superset.

Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:183]: oh wait, the text was actually updated on this one.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx (hunk 2 above)
Comment (betodealmeida) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:201]: Yeah, it doesn't make sense for other drivers. With psycopg2 SSL is enabled with `sslmode='require'`, but other driver have different approaches.

Comment (AAfghahi) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx]: yeah I agree, I ran it by Sophie and she agreed.

File: superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx (hunk 1 above)
Comment (hughhhh) [reply superset-frontend/src/views/CRUD/data/database/DatabaseModal/DatabaseConnectionForm.tsx:183]: Can we go with SSL is required here

**Comment by hughhhh** [comment]
LGTM, just fix the comment @betodealmeida mentioned for "SSL is required for Preset"

**Review by hughhhh — APPROVED** [review]

**Review by hughhhh — APPROVED** [review]

**Pull request merged**